src/Worker/Grpc/GrpcDurableTaskWorker.Processor.cs

```diff
@@ -186,10 +184,16 @@ async ValueTask<OrchestrationRuntimeState> BuildRuntimeStateAsync(
 
         async Task ProcessWorkItemsAsync(AsyncServerStreamingCall<P.WorkItem> stream, CancellationToken cancellation)
         {
+            // Create a new token source for timing out and a final token source that keys off of them both.
+            var timeoutSource = new CancellationTokenSource();
```

Comment:
```suggestion
            using CancellationTokenSource timeoutSource = new();
```

CTS with a timer will cause memory leaks if not disposed.

Reply:
Oops, fixed.